pippy/PipelineSchedule.py

```diff
@@ -56,12 +56,17 @@ def _maybe_get_loss(self, mb_index):
         else:
             return None
 
-    def _update_losses(self, losses):
+    def _update_losses(self, stages, losses):
         """
         Update the losses to those in the internal state
         """
+        # if stages not a list turn into a list
+        if not isinstance(stages, list):
+            stages = [stages]
+        contains_last_stage = any([stage.is_last for stage in stages])
+
```

Comment:
Because of this, maybe at least consider having two versions of this utility? One for single-stage schedules, the other for multi.